File: JavaArchitecture/app/src/main/java/com/demo/architecture/ui/main/MainContract.java
package com.demo.architecture.ui.main;


import com.demo.architecture.base.BasePresenter;
import com.demo.architecture.base.MvpView;
import com.demo.architecture.model.UpdateModel;

/**
 * Created by HeYingXin on 2017/6/23.
 */
public interface MainContract {
    interface View extends MvpView {
        public void updateApkSucc(UpdateModel updateModel);
    }

    abstract class MyPresenter extends BasePresenter<View> {
        public abstract void requestPermissions();
        public abstract void updateAPK();
    }
}

File: kotlin-frist-branch-TODO/app/src/main/java/com/mazouri/mvpkotlin/task/main/MainContract.kt
package com.mazouri.mvpkotlin.task.main

import android.support.annotation.StringRes
import com.mazouri.mvpkotlin.base.BasePresenter
import com.mazouri.mvpkotlin.base.IView
import com.mazouri.mvpkotlin.data.model.Repository

/**
 * Created by HeYingXin on 2018/4/16.
 */
object MainContract {

    interface View: IView {
        fun showOrganizations(repositories: MutableList<Repository>)
        fun showLoadReposFailed(error: String?)
        fun showError(error: String?)
        fun showError(@StringRes stringResId: Int)
    }

    abstract class Presenter: BasePresenter<View>() {
        abstract fun loadRepositories(repoUser: String?)
    }
}